pkg/impersonation/impersonation.go

```diff
@@ -323,3 +344,86 @@ func (i *Impersonator) waitForServiceAccount(sa *corev1.ServiceAccount) (*corev1
 	}
 	return ret, nil
 }
+
+func getUser(clusterContext *config.UserContext, userInfo user.Info, groupname string) (user.Info, error) {
+	u, err := clusterContext.Management.Management.Users("").Controller().Lister().Get("", userInfo.GetUID())
+	if err != nil {
+		return &user.DefaultInfo{}, err
+	}
+
+	groups := []string{"system:authenticated", "system:cattle:authenticated"}
+	if groupname != "" {
+		groups = append(groups, groupname)
```

Comment:
We're appending to groups and then using `insertSortedIfNotPresent` for it. Appending passed `groupname` may result in `groups` being no longer sorted and `insertSortedIfNotPresent` assumes passed slice is sorted.

Overall, I'm thinking if it would be simpler and less error-prone if we simply built slices by appending and then sorted and removed duplicates as the last steps (we could also do what `removeEmpty` does if applicable there too)? Just a thought, not necessary to change the implementation this close to code freeze.

Reply:
I decided to remove the insert function and simplified this